src/torrent.cpp

```diff
@@ -121,9 +124,12 @@ namespace libtorrent
 		}
 		return ret;
 	}
-
 	constexpr int default_piece_priority = 4;
 
+	// holds seed rank priority duration in the ratio type which is used at
+	// seed rank calculation to prevent multible conversion and constructions
+	static constexpr seconds m_seed_rank_priority_duration = seconds(30 * 60);
```

Comment:
Even if comparation operators are able to handle it, it would still calculate a value with same base every time instead of one time on construct.